backend/prisma/migrations/20251001090802_change_studentactivity_data_to_json/migration.sql

```diff
@@ -3,6 +3,10 @@
 
   - Changed the type of `data` on the `StudentActivityApp` table. No cast exists, the column would be dropped and recreated, which cannot be done if there is data, since the column is required.
 
+  Why this is okay:
+  The application is not yet in production, so there is no production data to worry about.
+  Furthermore, the application is still in active development, so any development data can be discarded. 
+
```

Comment:
I actually disagree with those reasons but agree that we can do this.

The application is under active development, yes, but there is also a deployed version that is being used by the customer. We cannot simply delete their data, they will be super unhappy about that! (The migration is automatically run when we deploy).

The reason why it is okay is that so far the `StudentActivityApp` has not been used and therefore cannot contain any data! If you want to be extra sure, you could also convert the data but I would say there is no need given the table has not been used yet.

But we did always migrate the data for the development deployment so far as well. You can see this in the other migration files.